_data-prepper/pipelines/configuration/processors/mutate-strings.md

```diff
@@ -0,0 +1,237 @@
+```
+
+If the `from` regex string does not return a match, the key returns without any changes.
+
+### Configuration
+
+<!--- Are these paramters? How are these items used? Need some intro text here.--->
+
+* `entries` (required): A list of entries to add to an event.
+    * `source` (required): The key to be modified.
+    * `from` (required): The regex string to be replaced. Special regex characters such as `[` and `]` must be escaped using `\\` when using double quotes and `\` when using single quotes. See [Class Pattern](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html) for more information.
+    * `to` (required): The string to be substituted for each match of `from`.
+    
+## split_string
+
+The `split_string` processor splits a field into an array using a delimiter character.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - split_string:
+        entries:
+          - source: "message"
+            delimiter: ","
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with your file path:
+
+```json
+{"message": "hello,world"}
+```
+When you run Data Prepper with this `pipeline.yaml` file, you should see the following output:
+
+```json
+{"message":["hello","world"]}
+```
+
+### Configuration
+
+* `entries` (required): A list of entries to add to an event
+* `source` (required): The key to be split
+* `delimiter` (optional): The separator character responsible for the split. Cannot be defined at the same time as `delimiter_regex`. At least `delimiter` or `delimiter_regex` must be defined.
+* `delimiter_regex` (optional): A regex string responsible for the split. Cannot be defined at the same time as `delimiter`. At least `delimiter` or `delimiter_regex` must be defined.
+
+## uppercase_string
+
+The `uppercase_string` processor converts a key to uppercase.
+
+<!--- Converts a key or a string? Are only keys and strings being converted?--->
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - uppercase_string:
+        with_keys:
+          - "uppercaseField"
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the correct file path:
+
+```json
+{"uppercaseField": "hello"}
+```
+When you run Data Prepper with this `pipeline.yaml` file, you should see the following output:
+
+```json
+{"uppercaseField": "HELLO"}
+```
+
+### Configuration
+
+<!--- Are these paramters? How are these items used? Need some intro text here.--->
```

Comment:
Same as previous Configuration sections. What does configuration specifically allow the user to do here?